src/lib/save-project-to-server.js

```diff
@@ -45,6 +45,7 @@ export default function (projectId, vmState, params) {
     return new Promise((resolve, reject) => {
         xhr(opts, (err, response) => {
             if (err) return reject(err);
+            if (response.statusCode !== 200) return reject(response.statusCode);
```

Comment:
My only note is that I find `return reject()` confusing, since you're not actually doing anything (as I understand it) with the return value -- this is just a shorthand for
```
reject(response.statusCode);
return;
```

(see https://stackoverflow.com/questions/33445415/javascript-promises-reject-vs-throw#comment71729644_38181613 )